extensions/http/src/main/java/org/mule/extension/http/api/request/HttpRequesterConfig.java

```diff
@@ -0,0 +1,281 @@
+
+    /**
+     * Defines if the request should be sent using streaming or not. If this attribute is not present,
+     * the behavior will depend on the type of the payload (it will stream only for InputStream). If set
+     * to true, it will always stream. If set to false, it will never stream. As streaming is done the request
+     * will be sent user Transfer-Encoding: chunked.
+     */
+    @Parameter
+    @Optional(defaultValue = "AUTO")
+    private Function<MuleEvent, HttpStreamingType> requestStreamingMode;
+
+    /**
+     * Defines if the request should contain a body or not. If AUTO, it will depend on the method (GET,
+     * HEAD and OPTIONS will not send a body).
+     */
+    @Parameter
+    @Optional(defaultValue = "AUTO")
+    private Function<MuleEvent, HttpSendBodyMode> sendBodyMode;
+
+    /**
+     * Maximum time that the request element will block the execution of the flow waiting for the HTTP response.
+     * If this value is not present, the default response timeout from the Mule configuration will be used.
+     */
+    @Parameter
+    @Optional
+    private Function<MuleEvent, Integer> responseTimeout;
+
+    /**
+     * If true, cookies received in HTTP responses will be stored, and sent in subsequent HTTP requests.
+     */
+    @Parameter
+    @Optional(defaultValue = "true")
+    @Expression(NOT_SUPPORTED)
+    private boolean enableCookies;
+
+    /**
+     * Specifies a RAML configuration file for the API that is being consumed.
+     */
+    @Parameter
+    @Optional
+    @Expression(NOT_SUPPORTED)
+    private RamlApiConfiguration ramlApiConfiguration;
+
+    @Inject
+    @DefaultTlsContextFactoryBuilder
+    private TlsContextFactoryBuilder defaultTlsContextFactoryBuilder;
+    @Inject
+    private MuleContext muleContext;
+    private CookieManager cookieManager;
+    private boolean stopped = false;
+
+    public String getName()
+    {
+        return configName;
+    }
+
+    @Override
+    public void initialise() throws InitialisationException
+    {
+        if (port == null)
+        {
+            port = muleEvent -> protocol.getDefaultPort();
+        }
+
+        if (protocol.equals(HTTP) && tlsContextFactory != null)
+        {
+            throw new InitialisationException(CoreMessages.createStaticMessage("TlsContext cannot be configured with protocol HTTP, " +
+                                                                               "when using tls:context you must set attribute protocol=\"HTTPS\""), this);
+        }
+
+        if (protocol.equals(HTTPS) && tlsContextFactory == null)
+        {
+            tlsContextFactory = defaultTlsContextFactoryBuilder.buildDefault();
+        }
+        if (tlsContextFactory != null)
+        {
+            LifecycleUtils.initialiseIfNeeded(tlsContextFactory);
+        }
+
+        if (enableCookies)
+        {
+            cookieManager = new CookieManager();
+        }
+    }
+
+    public Function<MuleEvent, String> getHost()
```

Comment:
are all of these getter consumed by anyone else? any chance you're exposing too much?